src/nspawn/nspawn-network.c

```diff
@@ -440,40 +445,91 @@ int check_interface(const char *name) {
         return 0;
 }
 
-int move_network_interfaces(pid_t pid, char **ifaces) {
+int open_network_namespace(pid_t pid) {
+        char path[STRLEN("/proc//ns/net") + DECIMAL_STR_MAX(pid_t) + 1];
+        int fd = -1;
+
+        xsprintf(path, "/proc/" PID_FMT "/ns/net", pid);
+
+        fd = open(path, O_CLOEXEC|O_RDONLY|O_NOCTTY);
+        if (fd < 0)
+                return log_error_errno(errno, "Failed to open network namespace %s: %m", path);
+
+        return fd;
+}
+
+int move_network_interfaces(int from, int to, char **ifaces) {
+        _cleanup_close_ int current_netns_fd = -1;
         _cleanup_(sd_netlink_unrefp) sd_netlink *rtnl = NULL;
         char **i;
         int r;
 
         if (strv_isempty(ifaces))
                 return 0;
 
-        r = sd_netlink_open(&rtnl);
+        /* We can only move network interfaces from the current network namespace so we enter the
+         * network namespace from which we want to move interfaces. */
+
+        char current_netns_path[STRLEN("/proc//ns/net") + DECIMAL_STR_MAX(pid_t) + 1];
+        xsprintf(current_netns_path, "/proc/" PID_FMT "/ns/net", getpid());
+
+        current_netns_fd = open(current_netns_path, O_CLOEXEC|O_RDONLY|O_NOCTTY);
+        if (current_netns_fd < 0)
+                return log_error_errno(errno, "Failed to open file %s: %m", current_netns_path);
+
+        r = namespace_enter(-1, -1, from, -1, -1);
```

Comment:
hmm, I'd prefer if we change namespaces only in child processes forked off precisely for that purpose, and which then dies. i.e. I am very strongly of the opinion that namespace changes should not be done willy-nilly and not jumping forth and back but just done as one-way street really. Hence, please use safe_fork() to fork something off that enters the namespace, does what it needs to do and then exits, and the parent just waits for that. I don#t wnat nspawn#s main process jumping namespaces really...

Reply:
I pushed a new version using `safe_fork`. Can you verify if `safe_fork` is invoked with the right flags?